lib/assets/gql-generator.js

```diff
@@ -23,11 +32,37 @@ var graphql = require('graphql'),
     const scalarTypes = ['String', 'ID', 'Boolean'];
     type = type.toString();
 
-    if (scalarTypes.includes(type)) {
-      return type;
+    let matchedType;
+
+    scalarTypes.every((scalarType) => {
+      if (type.includes(scalarType)) {
+        matchedType = scalarType;
+
+        return false;
+      }
+
+      return true;
+    });
+
+    if (matchedType) {
+      return matchedType;
     }
+    // If type is not from the sclar types but is a user defined type
     else if (type.includes('[')) {
-      type = type.split('[')[1].split(']')[0];
+      const getSplitString = (ArrayDepth, identifier) => {
+          let splitString = identifier;
+
+          while (ArrayDepth > 1) {
+            splitString += identifier;
+
+            ArrayDepth--;
+          }
+
+          return splitString;
+        },
+        depth = getDepthOfArray(type);
+
+      type = type.split(getSplitString(depth, '['))[1].split(getSplitString(depth, ']'))[0];
```

Comment:
@Akshay2402 Try to simplify this. i.e. we're nesting many calls and references in one sentence here. We can do it in steps.

Reply:
Done